### PR TITLE
Update dropdown select before emitting input event (closes #1366)

### LINKED ISSUE #1366: Value and selectedIndex are not valid in input event handler for dropdown select element
### Are you requesting a feature or reporting a bug?
Bug

### What is the current behavior?
The `value` and `selectedIndex` properties of a dropdown select are not set in `input` event handler.

### What is the expected behavior?
The `value`  and `selectedIndex` properties should be changed according to the selected child when executing `input` event handler.

### How would you reproduce the current behavior (if this is a bug)?
Run the test.

#### Provide the test code and the tested page URL (if applicable)

Tested page URL: https://andreybelym.github.io/test-pages/gh-1366.html

Test code

```js
import {ClientFunction} from 'testcafe';                                   
                                                                           
fixture `test`
    .page`https://andreybelym.github.io/test-pages/gh-1366.html`;

test(`choose value`, async t => {                                                                                                             
    await t                                                                
        .click(`select`)                                                   
        .click(`option[value=Nissan]`)                                     
        .expect(ClientFunction(() => window.selectValue)()).eql('Nissan'); 
});                                                                        
```

### Specify your

* operating system: Windows 10
* testcafe version: 0.14.0-alpha6
* node.js version: 7.4.0

### PR DESCRIPTION
**Pull request by AndreyBelym**

### CODE REVIEW

**Comment by AndreyBelym** [comment]
FPR. Tests are failed because of Saucelabs failure. \сс @AlexanderMoskovkin @helen-dikareva

File: src/client/automation/playback/click/select-child.js [hunk 1]
```diff
@@ -45,7 +45,10 @@ export default class SelectChildClickAutomation {
             this.childIndex = isOption ? domUtils.getOptionIndex(this.parentSelect, this.element) :
                               domUtils.getOptionGroupIndex(this.parentSelect, this.element);
 
-            this.clickCausesChange = isOption && !this.element.disabled && this.childIndex !== selectedIndex;
+            var parentOptGroup = domUtils.isOptionGroupElement(this.element.parentNode) ? this.element.parentNode : null;
```
Comment (helen-dikareva) [inline src/client/automation/playback/click/select-child.js]: Please check: can `select` have nested `optgroup` elements?

File: test/functional/fixtures/regression/gh-1366/pages/index.html [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+</head>
+
+<body>
+<select>
+    <option value="Ford">Ford</option>
+    <option value="Vauxhall">Vauxhall</option>
+    <option value="Nissan">Nissan</option>
+</select>
+<script>
+    document.querySelector('select').addEventListener('input', function (event){window.selectValue = event.target.value});
```
Comment (helen-dikareva) [inline test/functional/fixtures/regression/gh-1366/pages/index.html]: Issue contains information about wrong `value` and `selectedIndex`. Maybe we should add assertion of `selectedIndex` too.

File: src/client/automation/playback/click/select-child.js (hunk 1 above)
Comment (AndreyBelym) [inline src/client/automation/playback/click/select-child.js]: No, it's not possible, browser inserts closing `</optgroup>` tag automatically before the nesetd optgroup.

File: test/functional/fixtures/regression/gh-1366/pages/index.html (hunk 2 above)
Comment (AlexanderMoskovkin) [inline test/functional/fixtures/regression/gh-1366/pages/index.html]: I think it's redundant. 
The main point of the issue - wrong events order. We check there that the state of the `select` element is updated before `input` event, I believe it's enough to check `value` only

Comment (helen-dikareva) [inline test/functional/fixtures/regression/gh-1366/pages/index.html]: as you wish

**Pull request merged**